apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -9,18 +9,32 @@ import java.io.File
 open class GraphqlCompiler {
   private val moshi = Moshi.Builder().build()
 
-  fun write(relativePath: String) {
+  fun write(irFile: File) {
+    val packageName = irFile.absolutePath.formatPackageName()
     val irAdapter = moshi.adapter(QueryIntermediateRepresentation::class.java)
-    val ir = irAdapter.fromJson(File(relativePath).readText())
+    val ir = irAdapter.fromJson(irFile.readText())
     // TODO: Handle multiple or no operations
     val operation = ir.operations.first()
     val typeSpecBuilder = OperationTypeSpecBuilder(operation.operationName, operation.fields)
-    JavaFile.builder("test", typeSpecBuilder.build()).build()
-        .writeTo(OUTPUT_DIRECTORY.fold(File("build"), ::File))
+    JavaFile
+      .builder(packageName, typeSpecBuilder.build())
+      .build()
+      .writeTo(OUTPUT_DIRECTORY.fold(File("build"), ::File))
   }
 
   companion object {
     const val FILE_EXTENSION = "graphql"
     val OUTPUT_DIRECTORY = listOf("generated", "source", "apollo")
   }
+
+  private fun String.formatPackageName(): String {
+    val parts = split(File.separatorChar)
+    val srcFolderIndex = parts.indexOfFirst { it == "src" }
+    val graphqlFolderIndex = parts.indexOfFirst { it == "graphql" }
```

Comment:
Well, I guess the chances to have such case is very low. 
Sorry but by relative path what do you mean? Roll back to have `relativePath:String` as param instead of File?